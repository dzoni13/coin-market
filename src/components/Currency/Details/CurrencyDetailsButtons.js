import React, { Component }      from 'react';
import { Link }                  from 'react-router-dom';
import { Button, ButtonToolbar } from 'react-bootstrap';

class CurrencyDetailsButtons extends Component {
  render() {
    return (
      <ButtonToolbar>
        <Link to={'/'}>
          <Button
            className="link"
            bsStyle="link"
            > Back to list
          </Button>
        </Link>
        <Button
          className="refresh"
          onClick={this.props.getCurrencyDetails}
          > Refresh
        </Button>
      </ButtonToolbar>
    );
  }
}

export default CurrencyDetailsButtons;
