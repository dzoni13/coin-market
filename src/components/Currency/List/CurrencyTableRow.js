import React, { Component } from 'react';
import { Link }             from 'react-router-dom';

class CurrencyTableRow extends Component {

  render() {
    const currency = this.props.currency;
    const path = `/currency/${currency.id}/?convert=${currency.fiatCurrency}`;

    return (
      <tr className="currency-row">
        <td> {currency.rank} </td>
        <td>
          <Link to={path}> {currency.symbol} </Link>
        </td>
        <td> {currency.price} </td>
        <td> {currency.percentChange24h} </td>
      </tr>
    )
  }
}

export default CurrencyTableRow;
