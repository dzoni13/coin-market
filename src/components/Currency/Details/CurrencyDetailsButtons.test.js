import React                  from 'react';
import { shallow }            from 'enzyme';
import CurrencyDetailsButtons from './CurrencyDetailsButtons';
import CurrencyTableRow       from '../List/CurrencyTableRow';

const setup = () => {
  const props = {
    currency:
      {
        "availableSupply": null,
        "fiatCurrency": undefined,
        "id": "bitcoin",
        "lastUpdated": null,
        "marketCap": undefined,
        "name": "Bitcoin",
        "percentChange1h": null,
        "percentChange24h": null,
        "percentChange7d": null,
        "price": undefined,
        "priceBtc": null,
        "rank": "1",
        "symbol": "BTC",
        "totalSupply": null,
        "volume24h": undefined
      },
    getCurrencyDetails: jest.fn()
  }
  const currencyDetailsButtons = shallow(<CurrencyDetailsButtons  {...props}/>);
  const currencyTableRow = shallow(<CurrencyTableRow {...props}/>);
  return {
    props,
    currencyDetailsButtons,
    currencyTableRow
  }
}

describe('CurrencyDetailsButtons', () => {
  const { props, currencyDetailsButtons, currencyTableRow } = setup();

  it('the item should be refresh when click button', () => {
    currencyDetailsButtons.find('.refresh').simulate('click');
    expect(props.getCurrencyDetails).toBeCalled();
  });

  it('the user should be to `/` redirected when click button', () => {
    currencyDetailsButtons.find('.link').simulate('click');
    expect(currencyTableRow.find('.currency-row').exists()).toEqual(true);
  });
});
