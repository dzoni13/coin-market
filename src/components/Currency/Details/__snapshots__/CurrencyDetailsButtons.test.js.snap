// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyDetailsButtons renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrencyDetailsButtons
    currency={
        Object {
            "availableSupply": null,
            "fiatCurrency": undefined,
            "id": "bitcoin",
            "lastUpdated": null,
            "marketCap": undefined,
            "name": "Bitcoin",
            "percentChange1h": null,
            "percentChange24h": null,
            "percentChange7d": null,
            "price": undefined,
            "priceBtc": null,
            "rank": "1",
            "symbol": "BTC",
            "totalSupply": null,
            "volume24h": undefined,
          }
    }
    getCurrencyDetails={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "btn-toolbar",
      "children": Array [
        <Link
          replace={false}
          to="/"
>
          <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="link"
                    className="link"
                    disabled={false}
          >
                     Back to list
          </Button>
</Link>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="refresh"
          disabled={false}
          onClick={[Function]}
>
           Refresh
</Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="link"
            className="link"
            disabled={false}
>
             Back to list
</Button>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "link",
            "children": " Back to list",
            "className": "link",
            "disabled": false,
          },
          "ref": null,
          "rendered": " Back to list",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": " Refresh",
          "className": "refresh",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": " Refresh",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "btn-toolbar",
        "children": Array [
          <Link
            replace={false}
            to="/"
>
            <Button
                        active={false}
                        block={false}
                        bsClass="btn"
                        bsStyle="link"
                        className="link"
                        disabled={false}
            >
                         Back to list
            </Button>
</Link>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            className="refresh"
            disabled={false}
            onClick={[Function]}
>
             Refresh
</Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="link"
              className="link"
              disabled={false}
>
               Back to list
</Button>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "link",
              "children": " Back to list",
              "className": "link",
              "disabled": false,
            },
            "ref": null,
            "rendered": " Back to list",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": " Refresh",
            "className": "refresh",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " Refresh",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
